### PR TITLE
New Code: Qadir JavaScript Assignment 20 and 21 Done.

### PR DESCRIPTION
**Pull request by MKadeer17**

### CODE REVIEW

File: Qadir__Work/JavaScript/Assignment21/Question1/formValidation.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+function checkFormFieldsIsValidOrNot() {
+    event.preventDefault();
+
+    // UserName length 8 to 15
+    let userNamePattern = /^\w{8,15}$/;
+    let passwordPattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8})/i;
```
Comment (sanketmod) [inline Qadir__Work/JavaScript/Assignment21/Question1/formValidation.js:6]: Username must start with alphabet

File: Qadir__Work/JavaScript/Assignment21/Question1/Qadir__Assignment21__1.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                font-weight: 700;
+                font-size: 15px;
+            }
+        </style>
+        <title>Form Validation</title>
+    </head>
+    <body>
+        <center>
+            <div>
+                <h1>Login Page</h1>
```
Comment (sanketmod) [inline Qadir__Work/JavaScript/Assignment21/Question1/Qadir__Assignment21__1.html:49]: center tag is deprecated in html5 doc

File: Qadir__Work/JavaScript/Assignment21/Question2/Qadir__Assignment21__2.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                margin-top: 10px;
+                margin-left: 10px;
+            }
+        </style>
+        <title>Table Create, Update, Read and Delete</title>
+    </head>
+    <body>
+        <table id="richestManList">
+            <caption>List Of Richest Man in World</caption>
+            <tr>
```
Comment (sanketmod) [inline Qadir__Work/JavaScript/Assignment21/Question2/Qadir__Assignment21__2.html:48]: Wrong caption

**Review by sanketmod — APPROVED** [review]

File: Qadir__Work/JavaScript/Assignment21/Question1/formValidation.js (hunk 1 above)
Comment (MKadeer17) [reply Qadir__Work/JavaScript/Assignment21/Question1/formValidation.js:6]: Ok.

File: Qadir__Work/JavaScript/Assignment21/Question1/Qadir__Assignment21__1.html (hunk 2 above)
Comment (MKadeer17) [reply Qadir__Work/JavaScript/Assignment21/Question1/Qadir__Assignment21__1.html:49]: I know.

File: Qadir__Work/JavaScript/Assignment21/Question2/Qadir__Assignment21__2.html (hunk 3 above)
Comment (MKadeer17) [reply Qadir__Work/JavaScript/Assignment21/Question2/Qadir__Assignment21__2.html:48]: In Question They not given any table name or something.
Please read properly

File: Qadir__Work/JavaScript/Assignment20/Question1/Qadir__Assignment20__1__2.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <script src="./tableCrud.js"></script>
+        <style>
+            table,
```
Comment (Snehal-113) [inline Qadir__Work/JavaScript/Assignment20/Question1/Qadir__Assignment20__1__2.html:16]: the width of the table increases after clicking on add, try setting these properties for table
 margin-left: auto;
        margin-right: auto;
        table-layout: auto;
        width: 90%;

File: Qadir__Work/JavaScript/Assignment21/Question2/crudOperations.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        }
+        else {
+            throw new Error("Please Select Row for Delete")
+        }
+    }
+    catch(error) {
+        alert(error.message);
+    }
+}
```
Comment (Snehal-113) [inline Qadir__Work/JavaScript/Assignment21/Question2/crudOperations.js:126]: operations do not work as expected, after the records are deleted

**Review by Snehal-113 — CHANGES_REQUESTED** [review]

File: Qadir__Work/JavaScript/Assignment20/Question1/Qadir__Assignment20__1__2.html (hunk 4 above)
Comment (MKadeer17) [reply Qadir__Work/JavaScript/Assignment20/Question1/Qadir__Assignment20__1__2.html:16]: Yes Mam

File: Qadir__Work/JavaScript/Assignment21/Question2/crudOperations.js (hunk 5 above)
Comment (MKadeer17) [reply Qadir__Work/JavaScript/Assignment21/Question2/crudOperations.js:126]: Yes Mam.

**Review by poonam-khode — APPROVED** [review]

**Review by Snehal-113 — APPROVED** [review]

**Pull request merged**